Add show/hide field to schedule for jobs create modal

mlunoe opened the pull request:
---
~~⚠️ Related PR: https://github.com/mesosphere/reactjs-components/pull/329~~ Merged
~~⚠️ Updating reactjs-components needed for this to work properly https://github.com/dcos/dcos-ui/pull/960~~ Merged

---

![](https://cl.ly/0W3d1v0P3J0I/Screen%20Recording%202016-08-10%20at%2011.02.gif)

mlunoe commented:
Ping @daiyiwhale @wavesoft

wavesoft commented:
Sorry, accidental UI click to the github desktop 😨  reverting...

wavesoft commented:
I guess this is not in sync with #944 ? I had some troubles getting this to work with the new version of `reactjs-components`

wavesoft commented:
Also `startingDeadlineSeconds` value is lost when I switch to JSON.

mlunoe commented:
I will look into the `startingDeadlineSeconds`.

Not sure what you are referring to with this comment: https://github.com/dcos/dcos-ui/pull/948#issuecomment-239375890. This PR, as stated in the description, depended on a reactjs-components update, which landed. Did you do `npm i` before trying this out?

mlunoe commented:
Arh... my bad @wavesoft. I forgotten to push the rebased commits

mlunoe commented:
Good catch on the `startingDeadlineSeconds`! It should be transferred as expected now :)

orlandohohmeier commented on src/js/components/JobForm.js:
```diff
@@ -8,18 +8,33 @@ const METHODS_TO_BIND = [
+
+    this.validateForm();
+    // Update model before we update multipleDefinition
+    this.model = this.triggerTabFormSubmit();
```
Would be nice if we find a way to not store the `model` on the form itself. But as this pattern is also found in the SchemaForm I'm ok if you don't address it.

wavesoft commented:
Looks good to me!

orlandohohmeier commented:
Needs a rebase

mlunoe commented:
Ok, rebased. And also fixed a problem where the `forceUpdate` would cause focus to act weird changes in the labels not being transferred. The `forceUpdate` only happens when the enable schedule changes checkbox changes
